src/Aks/Aks/Commands/RunAzureRmAksCommand.cs

```diff
@@ -95,30 +97,46 @@ public class RunAzureRmAksCommand : KubeCmdletBase
 
         private string GetCommandContext()
         {
-            if (CommandFile == null || CommandFile.Length == 0)
+            if (CommandContextAttachment == null || CommandContextAttachment.Length == 0)
             {
                 return "";
             }
             List<string> filesToAttach = new List<string>();
-            string currentDirectory = SessionState.Path.CurrentFileSystemLocation.ToString();
-            if (CommandFile.Length == 1 && CommandFile[0].Equals("."))
+            string rootDirectory = SessionState.Path.CurrentFileSystemLocation.ToString();
+            if (CommandContextAttachment.Length == 1)
             {
-                WriteDebug(string.Format("Will zip all the files under {0}.", currentDirectory));
-                Directory.GetFiles(currentDirectory, "*", SearchOption.AllDirectories).ForEach(filepath =>
+                string commandContextAttachmentPath = Path.GetFullPath(Path.Combine(rootDirectory, CommandContextAttachment[0]));
+                if ((File.GetAttributes(commandContextAttachmentPath) & FileAttributes.Directory) == FileAttributes.Directory)
                 {
-                    filesToAttach.Add(filepath);
-                });
+                    rootDirectory = commandContextAttachmentPath;
+                    WriteDebug(string.Format("Will zip all the files under {0}.", rootDirectory));
+                    Directory.GetFiles(rootDirectory, "*", SearchOption.AllDirectories).ForEach(filepath =>
+                    {
+                        filesToAttach.Add(filepath);
+                    });
+                }
+                else
+                {
+                    filesToAttach.Add(commandContextAttachmentPath);
+                }
             }
             else
             {
-                foreach (var filepath in CommandFile)
+                foreach (var filepath in CommandContextAttachment)
                 {
-                    filesToAttach.Add(Path.Combine(currentDirectory, filepath));
+                    string commandContextAttachmentPath = Path.GetFullPath(Path.Combine(rootDirectory, filepath));
+                    if ((File.GetAttributes(commandContextAttachmentPath) & FileAttributes.Directory) == FileAttributes.Directory)
+                    {
+                        throw new AzPSArgumentException(
+                            Resources.NeedSameParentFolder,
+                            nameof(CommandContextAttachment));
+                    }
+                    filesToAttach.Add(commandContextAttachmentPath);
                 }
             }
             if (filesToAttach.Count == 0)
             {
-                WriteDebug("There is no files to attach!");
+                WriteWarning("There is no files to attach!");
```

Comment:
This is a warning so we should be more formal
```suggestion
                WriteWarning("Could not find any file to attach. The command will be run without attachments.");
```